### PR TITLE
BENCH-70-compile and build answerking app using github-actions

### PR DESCRIPTION
**Pull request by JoeCSykes**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (PietroConvalleAD) [inline .github/workflows/ci.yml:12]: is it the latest LTS (long time support) or just latest release?

**Comment by PietroConvalleAD** [comment]
`/home/runner/work/AnswerKing-Python/AnswerKing-Python/answerking_python does not contain any element
Error: Process completed with exit code 1.`

we have this error from some package during install but the install is actually alright. we have to find a workaround.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      with:
+        python-version: '3.10'
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install poetry
+        poetry install
+    - name: Run Tests
+      run: |
+        python manage.py test
```
Comment (PietroConvalleAD) [inline .github/workflows/ci.yml]: `poetry run python manage.py test`?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install poetry
-        poetry install
+        poetry update
```
Comment (PietroConvalleAD) [inline .github/workflows/ci.yml:43]: why this change?

**Comment by PietroConvalleAD** [comment]
mysql set up is missing.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (PietroConvalleAD) [reply .github/workflows/ci.yml:43]: oh, well, I see fixes the error that we were receiving, not sure why and how but ok cool

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JoeCSykes) [reply .github/workflows/ci.yml:12]: I think it currently runs 18.04 LTS but will be switching to 20.04 LTS on Nov 30th (see https://github.blog/changelog/2020-10-29-github-actions-ubuntu-latest-workflows-will-use-ubuntu-20-04/). Seen as there is 22.04 LTS I think it runs the latest ubuntu LTS that they have tested against github workflows. As such I think were ok with just putting `ubuntu-latest`

**Comment by JoeCSykes** [comment]
Need to change DBPWD to equal a GitHub secret but currently don't have permissions to create a secret on this project. I have messaged Oleksii to see whether he wants to give me permissions or whether he wants me to do it another way. I feel like using secrets is the safest way to do this.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (PietroConvalleAD) [reply .github/workflows/ci.yml:12]: that article is from 2020, https://github.com/actions/runner-images#available-images looks like it's the 22.04 LTS 👍

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - name: Set up MySQL
+      uses: mirromutth/mysql-action@v1.1
+      with: 
+        host port: 3306
+        mysql database: 'AnswerKingBDTest'
+        mysql root password: "UnsafePassword098"
+  
+  
+  test:
+    runs-on: ubuntu-latest
```
Comment (PietroConvalleAD) [inline .github/workflows/ci.yml]: My guess is that it's failing because you are redefining the environment and hence you don't have the dependencies installed

**Comment by PietroConvalleAD** [comment]
needs linting and typing check

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -46,6 +46,9 @@ jobs:
     - name: Run Tests     
       run: |
         poetry run python manage.py test
+        poetry run black . --line-length=79
```
Comment (PietroConvalleAD) [inline .github/workflows/ci.yml]: I think we want to run `poetry run black . --line-length=79 --check` since we are just looking if the files are formatted properly

**Review by PietroConvalleAD — DISMISSED** [review]
Looks good to me, the only think left is using GH secrets

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
+name: CI
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
```
Comment (abbas-khan8) [inline .github/workflows/ci.yml]: Do we want to add main to this?

**Review by abbas-khan8 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 6 above)
Comment (JoeCSykes) [reply .github/workflows/ci.yml]: can do if people want?

**Comment by PietroConvalleAD** [comment]
Needs rebase with development now

File: .github/workflows/ci.yml (hunk 6 above)
Comment (JoeCSykes) [reply .github/workflows/ci.yml]: added main

**Review by PietroConvalleAD — APPROVED** [review]

File: answerking_app/utils/mixins/CategoryItemMixins.py [hunk 7]
```diff
@@ -12,7 +12,7 @@
 
 class CategoryItemUpdateMixin:
     def update(
-        self, request: Request, cat_id: int, item_id: int, *args, **kwargs
+        self, request: Request, cat_id: str, item_id: int, *args, **kwargs
```
Comment (PietroConvalleAD) [inline answerking_app/utils/mixins/CategoryItemMixins.py]: why?

Comment (PietroConvalleAD) [reply answerking_app/utils/mixins/CategoryItemMixins.py]: oh you are testing the fails, nvm
